src/merge.c

```diff
@@ -3255,6 +3255,10 @@ int git_merge(
 		&checkout_strategy)) < 0)
 		goto done;
 
+	if ((error = git_repository_index(&index, repo) < 0) ||
+	    (error = git_index_read(index, 0) < 0))
```

Comment:
We're leaking this index now.  I would encourage us to use a separate variable here, eg:

```
if ((error = git_repository_index(&repo_index, repo) < 0) ||
    (error = git_index_read(repo_index, 0) < 0))
```

And be sure to `git_index_free` this `repo_index`.